### DISCUSSION TITLE
XTTS v2 fine-tune the model in FP16

### DISCUSSION THREAD

**Discussion by HobisPL** [coqui-ai/TTS#3316] [General Q&A]
What else do I need to add to ensure proper FP16 training acceleration?
When I change mixed_precision=True, all the losses become (nan):
![Zrzut ekranu 2023-11-26 230558](https://github.com/coqui-ai/TTS/assets/95121145/8da81144-6458-4c6a-a6f0-f173e11292a6)

fp32 displays normally, but the training process is very slow..
![Zrzut ekranu 2023-11-26 225634](https://github.com/coqui-ai/TTS/assets/95121145/63cc4e48-c28e-465e-8fdd-114c56f09ed9)

**Comment by HobisPL** [comment]
nobody knows ? o.O

**Reply by erogol** [reply]
you can get None with fp16 when the original model is not trained with it. You can try starting from very small LRs
